validator/rpc/wallet.go

```diff
@@ -165,6 +133,80 @@ func (s *Server) WalletConfig(ctx context.Context, _ *empty.Empty) (*pb.WalletRe
 	}, nil
 }
 
+// RecoverWallet via an API request, allowing a user to recover a derived.
+// Generate the seed from the mnemonic + language
+// Create N validator keystores from the seed specified by req.NumAccounts
+// Set the wallet password to req.WalletPassword (needs password validation)
+// Create the wallet
+// Return &CreateWalletResponse, nil if nothing went wrong
+func (s *Server) RecoverWallet(ctx context.Context, req *pb.RecoverWalletRequest) (*pb.CreateWalletResponse, error) {
```

Comment:
This function needs test

Reply:
This functions was testing using an adaptation of the func [TestServer_CreateWallet_Derived](https://github.com/prysmaticlabs/prysm/blob/4886a4e749b16adb1725004cee39661c6e10f00e/validator/rpc/wallet_test.go#L86)
The Test func was adapted to handle passing a language, mnemonic, skip25pass flag and the passphrase.

Reply:
Did i miss other scenarios ?

Reply:
The following scenarios are missed, it'll be great to cover them
1.) `password did not pass validation`
2.) `Could not write wallet password to disk`

Reply:
First password test added and passed, since it was new code. 
As for the write wallet password to disk. Glad you pointed it out.  The already existing Create Imported does the same . So adding,  at the end , the following lines(taken from the [Test_writeWalletPasswordtoDisk )](url) to both Recover (derived) and Create Imported  causes the test to fail.  Need to investigate. 


```
       // File should have been written.
	assert.Equal(t, true, fileutil.FileExists(localWalletDir))

	// Attempting to write again should trigger an error.
	err = writeWalletPasswordToDisk(localWalletDir, "somepassword")
	require.NotNil(t, err)
```

Reply:
So after @rauljordan  pointed out the flag i added it to the test case. But Recover is still not writing. 
I update the flag to True and expect RecoverWallet to write the password,  but it does not. 
Then a defer call to set to False.

Reply:
Awesome. We just need a test for the following error condition then we should be good to go:
```go
validatingPublicKeys, err := km.FetchValidatingPublicKeys(ctx)
	if err != nil {
		return errors.Wrap(err, "could not check for validating public keys")
	}
```

Reply:
@terencechain  this piece of code was not added (pre this PR) and is already existed  tested in [TestServer_ImportKeystores_OK](https://github.com/ahadda5/prysm/blob/2bdb65e668d4cc205303ef4b0099bc69f59395a8/validator/rpc/wallet_test.go#L257)   It initializes a wallet, a child keymanager and fills it with keystores. It fetches before importing the keystore, then fetches the public keys again.
  Is there a something specific you would like me to add?

Reply:
The following error condition is not tested according to code coverage though:
```
could not check for validating public keys
```

I think that something that we can add with relatively low effort